src/core_ct/core.py

```diff
@@ -106,7 +106,131 @@ def trim(self, axis: int, loc_start: int, loc_end: int | None = None) -> None:
                     :, :, loc_start : len(self.pixel_array[0, 0]) - loc_end
                 ]
             case _:
-                raise Exception("axis must be a value between 0 and 2 (inclusive)")
+                raise ValueError("axis must be a value between 0 and 2 (inclusive)")
+
+    def swapaxes(self, axis1: int, axis2: int) -> None:
+        """
+        Swap two axes of pixel_array and update pixel_dimensions accordingly.
+
+        Arguments:
+        ---------
+            axis1: integer either 0,1,2 specifying one axis to swap:
+                    0 corresponds to x-axis
+                    1 corresponds to y-axis
+                    2 corresponds to z-axis
+            axis2: integer either 0,1,2 specifying the second axis to swap:
+                    0 corresponds to x-axis
+                    1 corresponds to y-axis
+                    2 corresponds to z-axis
+
+        Returns:
+        -------
+            None
+
+        Raises:
+        ------
+            ValueError if axes are values other than 0, 1, or 2
+        """
+        # make sure axis inputs are valid
+        if axis1 < 0 or axis1 > 2:
+            raise ValueError("axis1 must be a value between 0 and 2 (inclusive)")
+        if axis2 < 0 or axis2 > 2:
+            raise ValueError("axis2 must be a value between 0 and 2 (inclusive)")
+        
+        # swap axes in pixel array
+        self.pixel_array = np.swapaxes(self.pixel_array, axis1, axis2)
+
+        # swap values in pixel dimensions
+        temp = self.pixel_dimensions[axis1]
+        self.pixel_dimensions[axis1] = self.pixel_dimensions[axis2]
+        self.pixel_dimensions[axis2] = temp
+
+    def flip(self, axis: int) -> None:
+        """
+        Flips pixel_array on a given axis.
```

Comment:
Can you add more detail to the docstring about what it means to "flip" the pixel array on an axis? I assume it means it is just reversing the array at that dimension, but it would be nice to specify that since the differences between rotating, flipping, and swapping axes might not be immediately obvious.

Reply:
I reworded it to be "reversed along the given access"